.github/workflows/tests.yml

```diff
@@ -40,10 +40,39 @@ jobs:
         if: github.event_name == 'pull_request'
         id: formulae-detect
 
+  timeout_labels:
+    if: github.event_name == 'pull_request' && github.repository == 'Homebrew/homebrew-core'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Count pull requests with the `CI-long-timeout` label
+        if: contains(github.event.pull_request.labels.*.name, 'CI-long-timeout')
+        uses: actions/github-script@v5
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const query = `query($owner:String!, $name:String!, $label:String!) {
+              repository(owner:$owner, name:$name) {
+                pullRequests(last:100, states:OPEN, labels: [$label]) {
+                  totalCount
+                }
+              }
+            }`;
+            const variables = {
+              owner: context.repo.owner,
+              name: context.repo.repo,
+              label: 'CI-long-timeout'
+            }
+            const { repository: { pullRequests: { totalCount } } }  = await github.graphql(query, variables)
+            core.setOutput('count', totalCount)
+
+            if (totalCount > 2) {
+              core.setFailed(`Too many pull requests (${totalCount}) with the long-timeout label!`)
+            }
+
   setup_tests:
     if: github.event_name == 'pull_request' && github.repository == 'Homebrew/homebrew-core'
     runs-on: ubuntu-latest
-    needs: tap_syntax
+    needs: [tap_syntax, timeout_labels]
```

Comment:
Perhaps this could be part of `tap_syntax` as it's already doing label handling?

Reply:
Yep, makes sense. I'd been focusing on getting something working first before trying to figure out how to integrate this into the rest of the workflow.